### PR TITLE
Support pyarrow=6.*

### PR DESCRIPTION
**Pull request by xhochy**
# Description:

Briefly describe the change of behavior

- [ ] Closes #xxxx
- [ ] Changelog entry

### CODE REVIEW

**Comment by xhochy** [comment]
All green except the nightlies. I will work on the added limitations in separate PRs.

**Review by jtilly — APPROVED** [review]

**Comment by DamianBarabonkovQC** [comment]
Is `CI / build (true, ubuntu-latest, 4.0.1, 3.8)` a nightly?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -116,7 +120,8 @@ jobs:
 
       # Tests
       - name: Pytest
-        run: pytest -n auto --dist loadscope --cov --cov-report xml
+        # FIXME: Add `-n auto` again to allow for parallelism here.
+        run: pytest --cov --cov-report xml
```
Comment (DamianBarabonkovQC) [inline .github/workflows/ci.yml:124]: From experience, the `pytest-cov` runs after module initialization. So parts of any imported module that run directly after import will happen before the coverage and thus won't be a part of the coverage. I used `coverage run pyest ...`.

File: docs/environment-docs.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ dependencies:
   # Currently dask and numpy==1.16.0 clash
   - numpy!=1.15.0,!=1.16.0
   - pandas>=0.23.0, !=1.0.0
-  - pyarrow>=0.17.1,!=1.0.0, <4
+  - pyarrow>=0.17.1,!=1.0.0, <6
```
Comment (DamianBarabonkovQC) [inline docs/environment-docs.yml:11]: Shouldn't this be `<7`?

Comment (xhochy) [reply docs/environment-docs.yml:11]: Yes, it should!

**Pull request merged**